pkg/identityserver/end_device_registry.go

```diff
@@ -188,13 +227,33 @@ func (is *IdentityServer) getEndDevice(ctx context.Context, req *ttnpb.GetEndDev
 		defer func() { is.setFullEndDevicePictureURL(ctx, dev) }()
 	}
 
+	if ttnpb.HasAnyField(ttnpb.TopLevelFields(req.FieldMask.GetPaths()), "claim_authentication_code") {
```

Comment:
I think `HasAnyField` already considers top level fields, doesn't it?

Reply:
Indeed the use of `ttnpb.TopLevelFields` in here might not be strictly necessary.
Considering that the values would be something like:
```
"claim_authentication_code",
"claim_authentication_code.valid_from",
"claim_authentication_code.valid_to",
"claim_authentication_code.value"
```

Doesn't the ttnpb.HasAnyField` be enough for the validation? @KrishnaIyer

Reply:
> I think HasAnyField already considers top level fields

It does not and hence the use of `ttnpb.TopLevelFields`. This is explicitly specified in the GoDoc
```
...
// Note that this function may have unexpected results when non bottom search fields are used,
// as HasAnyField([]string{"a.b"}, "a") is false.
...
func HasAnyField(requested []string, search ...string) bool
```

Without explicitly using `ttnpb.TopLevelFields`, the following command will retrieve nothing.
```
./ttn-lw-cli dev get test-app test-dev --claim-authentication-code.secret
```
whereas the following will work
```
./ttn-lw-cli dev get test-app test-dev --claim-authentication-code
```
As explained in the other comment, this is not for validation but for the fact that the gormstore implementation requires the top-level field `claim_authentication_code` to get/set any of the sub-fields. This check can be removed when we fully migrate to bun.

Reply:
True, my bad on missing the GoDoc info. Then this looks appropriate for me.